### PR TITLE
fixup paths in kubernetes_e2e.py to make less assumptions

### PR DESCRIPTION
**Pull request by BenTheElder**
this cleans up more path assumptions, ref: https://github.com/kubernetes/test-infra/issues/4724 and https://github.com/kubernetes/test-infra/issues/4724

### CODE REVIEW

**Comment by BenTheElder** [comment]
:bike: :houses: :paintbrush: 
/shrug

File: scenarios/kubernetes_e2e.py [hunk 1]
```diff
@@ -575,7 +574,7 @@ def main(args):
     if args.kubeadm:
         version = kubeadm_version(args.kubeadm, shared_build_gcs_path)
         runner_args.extend([
-            '--kubernetes-anywhere-path=/workspace/kubernetes-anywhere',
```
Comment (ixdy) [inline scenarios/kubernetes_e2e.py]: format substitution would be more consistent with the following lines

Comment (BenTheElder) [reply scenarios/kubernetes_e2e.py]: done.

**Comment by ixdy** [comment]
I'm a little worried that not overriding $HOME is going to cause different failures.

but let's see what happens!
/lgtm
:shipit:

**Comment by BenTheElder** [comment]
More world watching @krzyzacy :smile:

File: scenarios/kubernetes_e2e.py [hunk 2]
```diff
@@ -128,7 +128,6 @@ def __init__(self, workspace, artifacts):
         self.os_env = []
         self.env_files = []
         self.add_environment(
-            'HOME=%s' % workspace,
```
Comment (krzyzacy) [inline scenarios/kubernetes_e2e.py]: https://github.com/kubernetes/test-infra/blob/master/kubetest/main.go#L659 :joy: :joy: :joy:

Comment (BenTheElder) [reply scenarios/kubernetes_e2e.py]: AHHHHH

Comment (ixdy) [reply scenarios/kubernetes_e2e.py]: ITYM
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

**Comment by ixdy** [comment]
we can probably roll-forward this change without the `$HOME` change, right?

**Comment by BenTheElder** [comment]
> we can probably roll-forward this change without the $HOME change, right?

Yeah, that's the conclusion I reached with @krzyzacy before I caught up with my notifications :-) --  I'll do that.

**Pull request merged**